kernel/pte_flags/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+//! This crate defines the structure of page table entry (PTE) flags on x86_64 and aarch64.
+//! 
+//! For aarch64, the definition of these flags assumed that the [MAIR] index 0
+//! has a "DEVICE nGnRE" entry, and [MAIR] Index 1 has a Normal + Outer Shareable entry.
+//!
+//! [MAIR]: https://docs.rs/cortex-a/latest/cortex_a/registers/MAIR_EL1/index.html
+
+#![no_std]
+
+use cfg_if::cfg_if;
+use bitflags::bitflags;
+
+cfg_if!{ if #[cfg(target_arch = "x86_64")] {
+    mod pte_flags_x86_64;
+    pub use pte_flags_x86_64::PteFlagsX86_64;
+    pub use pte_flags_x86_64::PteFlagsX86_64 as PteFlagsArch;
+} else if #[cfg(target_arch = "aarch64")] {
+    mod pte_flags_aarch64;
+    pub use pte_flags_aarch64::PteFlagsAarch64;
+    pub use pte_flags_aarch64::PteFlagsAarch64 as PteFlagsArch;
+}}
+
+bitflags! {
+    /// Common, architecture-independent flags for a page table entry (PTE)
+    /// that define how a page is mapped.
+    ///
+    /// **Note:** items beginning with an underscore `_` are not used in Theseus.
+    ///
+    /// Also, this contains only the flags that are common to both `x86_64` and `aarch64`.
+    ///
+    /// This type can be losslessly converted into `PteFlagsX86_64` and `PteFlagsAarch64`
+    /// with the typical [`From`] and [`Into`] traits.
+    /// This makes it to set general architecture-indepedent flags first,
+    /// and then convert it in order to set more architecture-specific flags.
+    /// 
+    /// This type can also be converted *from* `PteFlagsX86_64` and `PteFlagsAarch64`,
+    /// but it may be lossless as only the bit flags defined herein are preserved.
+    pub struct PteFlags: u64 {
```

Comment:
I think it lacks a way to set PAGE_DESCRIPTOR & ACCESSED; I needed to set this bit for my descriptors to be used correctly by qemu's emulated CPU. As I wrote below, the CPU refused to use the descriptors I was creating if they didn't have the ACCESSED bit set.

Reply:
hmm, i could add a function for setting ACCESSED, but for the PAGE_DESCRIPTOR that's something you should need to manually set by doing something like `flags.insert( PteFlagsAarch64::PAGE_DESCRIPTOR)` or remove, etc.